library/azure_rm_devtestlabvirtualmachine.py

```diff
@@ -0,0 +1,533 @@
+        - absent
+        - present
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Create (or update) Virtual Machine
+    azure_rm_devtestlabvirtualmachine:
+      resource_group: myrg
+      lab_name: mylab
+      name: myvm
+      notes: Virtual machine notes....
+      os_type: linux
+      vm_size: Standard_A2_v2
+      user_name: vmadmin
+      password: ZSuppas$$21!
+      lab_subnet:
+        name: myvnSubnet
+        virtual_network_name: myvn
+      disallow_public_ip_address: no
+      image:
+        offer: UbuntuServer
+        publisher: Canonical
+        sku: 16.04-LTS
+        os_type: Linux
+        version: latest
+      artifacts:
+        - source_name: myartifact
+          source_path: "/Artifacts/linux-install-mongodb"
+      allow_claim: no
+      expiration_date: "2019-02-22T01:49:12.117974Z"
+'''
+
+RETURN = '''
+id:
+    description:
+        - The identifier of the DTL Virtual Machine resource.
+    returned: always
+    type: str
+    sample: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourcegroups/myrg/providers/microsoft.devtestlab/labs/mylab/virtualmachines/myvm
+compute_id:
+    description:
+        - The identifier of the underlying Compute Virtual Machine resource.
+    returned: always
+    type: str
+    sample: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourcegroups/myrg/providers/microsoft.devtestlab/labs/mylab/virtualmachines/myvm
+fqdn:
+    description:
+        - Fully qualified domain name or IP Address of the virtual machine.
+    returned: always
+    type: str
+    sample: myvm.eastus.cloudapp.azure.com
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+from ansible.module_utils.common.dict_transformations import _snake_to_camel
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrest.polling import LROPoller
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.devtestlabs import DevTestLabsClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMVirtualMachine(AzureRMModuleBase):
+    """Configuration class for an Azure RM Virtual Machine resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            lab_name=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            notes=dict(
+                type='str'
+            ),
+            os_type=dict(
+                type='str',
+                choices=['linux', 'windows'],
+                default='linux'
```

Comment:
is os_type updatable?  if yes, then default value will introduce some trouble as you know, when user not setting os_type at second run..

Reply:
no it's not